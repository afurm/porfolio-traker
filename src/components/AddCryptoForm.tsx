import React, { useState, useEffect } from 'react';
import { getCryptoList } from '@/api/cryptoService';
import { TextField, Button, Box, ToggleButtonGroup, ToggleButton, List, ListItem, ListItemButton, ListItemText } from '@mui/material';

interface Crypto {
  id: string;
  symbol: string;
  name: string;
}

const AddCryptoForm = () => {
  const [coinName, setCoinName] = useState<string>('');
  const [amount, setAmount] = useState<number>(0);
  const [transactionType, setTransactionType] = useState<'buy' | 'sell'>('buy');
  const [suggestions, setSuggestions] = useState<Crypto[]>([]);
  const [cryptoList, setCryptoList] = useState<Crypto[]>([]);

  const handleUpdatePortfolio = async (coinName: string, amount: number, type: 'buy' | 'sell') => {
 
  };

  useEffect(() => {
    const fetchCryptoList = async () => {
      const data = await getCryptoList();
      setCryptoList(data);
    };
    fetchCryptoList();
  }, []);

  const handleToggleTransactionType = (event: React.MouseEvent<HTMLElement>, newType: 'buy' | 'sell' | null) => {
    if (newType !== null) {
      setTransactionType(newType);
    }
  };

  const handleCoinNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value;
    setCoinName(value);

    if (value.length > 0) {
      const filteredSuggestions = cryptoList.filter((coin) =>
        coin.name.toLowerCase().includes(value.toLowerCase())
      );
      setSuggestions(filteredSuggestions);
    } else {
      setSuggestions([]);
    }
  };

  const handleSelectSuggestion = (suggestion: string) => {
    setCoinName(suggestion);
    setSuggestions([]);
  };

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    if (coinName && amount > 0) {
      handleUpdatePortfolio(coinName, amount, transactionType);
      setCoinName('');
      setAmount(0);
      setSuggestions([]);
    }
  };

  return (
    <Box component="div" sx={{ padding: 4, maxWidth: 600, margin: '0 auto', backgroundColor: '#ffffff', borderRadius: 2, boxShadow: 3 }}>
      <form onSubmit={handleSubmit}>
        <ToggleButtonGroup
          value={transactionType}
          exclusive
          onChange={handleToggleTransactionType}
          sx={{ marginBottom: 2 }}
        >
          <ToggleButton value="buy" aria-label="buy">
            Buy
          </ToggleButton>
          <ToggleButton value="sell" aria-label="sell">
            Sell
          </ToggleButton>
        </ToggleButtonGroup>
        <TextField
          label="Coin Name"
          value={coinName}
          onChange={handleCoinNameChange}
          fullWidth
          sx={{ marginBottom: 2 }}
        />
        {suggestions.length > 0 && (
          <List>
            {suggestions.slice(0, 10).map((suggestion, index) => (
              <ListItem key={index} disablePadding>
                <ListItemButton onClick={() => handleSelectSuggestion(suggestion.name)}>
                  <ListItemText primary={`${suggestion.name} (${suggestion.symbol.toUpperCase()})`} />
                </ListItemButton>
              </ListItem>
            ))}
          </List>
        )}
        <TextField
          label="Amount"
          type="number"
          value={amount}
          onChange={(e) => setAmount(Number(e.target.value))}
          fullWidth
          sx={{ marginBottom: 2 }}
        />
        <Button type="submit" variant="contained" color="primary" fullWidth>
          Submit Transaction
        </Button>
      </form>
    </Box>
  );
};

export default AddCryptoForm;