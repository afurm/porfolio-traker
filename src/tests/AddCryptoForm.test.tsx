import React from 'react';
import { render } from '@testing-library/react';

import '@testing-library/jest-dom';
import AddCryptoForm from '@/components/AddCryptoForm';

describe('AddCryptoForm Snapshot Test', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<AddCryptoForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
