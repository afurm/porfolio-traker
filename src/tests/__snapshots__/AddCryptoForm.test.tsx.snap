// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCryptoForm Snapshot Test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-10ezmbe"
  >
    <form>
      <div
        class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal css-1xcmbr1-MuiToggleButtonGroup-root"
        role="group"
      >
        <button
          aria-label="buy"
          aria-pressed="true"
          class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-firstButton css-4y8j39-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="buy"
        >
          Buy
        </button>
        <button
          aria-label="sell"
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-lastButton css-4y8j39-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="sell"
        >
          Sell
        </button>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-itbxsg-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1q964xs-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Coin Name
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1o0kqxp-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1pbc52w"
            >
              <span>
                Coin Name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-itbxsg-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-18pjc51-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r1:"
          id=":r1:-label"
        >
          Amount
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1o0kqxp-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            type="number"
            value="0"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-kg5swy"
            >
              <span>
                Amount
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1yujjup-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Submit Transaction
      </button>
    </form>
  </div>
</DocumentFragment>
`;
